modules/csv/src/csv-loader.js

```diff
@@ -63,7 +63,8 @@ function parseCSVInBatches(asyncIterator, options) {
 
       // Check if we need to save a header row
       if (isFirstRow && !headerRow) {
-        if (isHeaderRow(row)) {
+        const {header = isHeaderRow(row)} = options;
```

Comment:
Total nit: Just a bit too clever for my taste (i.e. optimize for readability not lines of code). I already felt the header handling was more complex than I would have preferred...

Would something like this work? 
```
if (options.header || isHeaderRow())
```